tests/performance-test/grafana/grafana-launcher.sh

```diff
@@ -19,11 +19,37 @@ OCP_PASS=$(oc get secret -n openshift-monitoring grafana-datasources -o jsonpath
     python -c "import sys, json; print json.load(sys.stdin)['datasources'][0]['basicAuthPassword']")
 sed -e "s|<<OCP_PASS>>|${OCP_PASS}|g" ./datasource.yaml > /tmp/datasource.yaml
 
-oc create configmap grafana-config --from-file ../grafana/grafana.ini
+oc create configmap grafana-config --from-file grafana.ini
 oc create configmap datasources-config --from-file /tmp/datasource.yaml
-oc create -f ../grafana/grafana-service.yml
-oc create -f ../grafana/grafana-route.yml
-oc create -f ../grafana/grafana-deploy.yml
+oc create -f grafana-service.yml
+oc create -f grafana-route.yml
+oc create -f grafana-deploy.yml
+
+# First the deployment will say it's Running!
+oc rollout status deployment/grafana-deployment
+
+# Then the pod will say it's ready (it will be a lie)
+while oc get pod -l app=grafana -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}' | grep False; do
+    oc get pod -l app=grafana
+    sleep 3
+done
+
+# Finally the logs will say it's listening - Hi Grafana!
+pod=$(oc get pod -l app=grafana -o jsonpath='{.items[0].metadata.name}')
+echo -n "Waiting for grafana to be listening"
+while ! oc logs $pod | grep -o "HTTP Server Listen"; do echo ...; done
+
+if ! GRAF_HOST=$(oc get routes --field-selector metadata.name=grafana -o jsonpath="{.items[0].spec.host}") 2> /dev/null; then 
+    echo "Error: cannot find Grafana instance in cluster." 1>&2
+    exit 1
+fi
+
+printf "\n*** Creating new dashboards in Grafana ***\n"
```

Comment:
This works, although a more elegant solution would probably be to utilize Grafana's [dashboard provisioning directory](https://grafana.com/docs/administration/provisioning/#dashboards) (the same way data sources are done above).

Reply:
Ah, thanks, I was looking for that but ended up just moving the existing code instead. I'll give that a shot!